### PR TITLE
Convert inline HTML style attributes

### LINKED ISSUE #1473: handle HTML in MD cells for tutorials
### Issue Description
The notebook to mdx converter does not currently handle HTML in markdown cells. The
initial reason why we did not handle HTML in MD cells is because the user can write
style tags in the cell, which React & Docusaurus will not parse. This is why the
tutorials thus far been using style attributes in div tags. See `Semantics for
@bm.random_variable` text block as an example.

This leads to a disconnect with what a tutorial writer sees in a Jupyter environment,
and what is seen in the Docusaurus build. See the below images as an example.

*No formatting in Jupyter*
no-formatting-jupyter

*Formatting in Docusaurus*
with-formatting-docusaurus

### Steps to Reproduce
All current tutorials exhibit this behavior.

### Expected Behavior
Proper handling of HTML components in MD that convert properly to MDX.

### PR DESCRIPTION
**Pull request by ndmlny-qs**
### Motivation

We currently do not handle HTML in Markdown as any style attributes will
not be formatted correctly for React. This change uses `mdformat` in
combination with finding inline style attributes that are reformatted so
React can consume them.

The change allows a user to write HTML in a Jupyter session with inline
styles applied to any HTML tag they wish. Those styles will be converted
to a React appropriate object that Docusaurus can render correctly. Thus
allowing the user to see the same styles being applied to both their
Jupyter session and in a Docusaurus build.

Resolves #1473

### Changes proposed

- Updates to the `convert_ipynb_to_mdx.py` script;
  - search and find all inline `script` attributes, and reformat them;
  - fix deprecation warning about showing the index when a pandas
    DataFrame is saved to Markdown.
- Updates to notebooks that were previously using React style `style`
  values.
  - Coin_flipping.ipynb
  - GMM_with_2_dimensions_and_4_components.ipynb
  - Hidden_Markov_model.ipynb
  - Hierarchical_modeling.ipynb
  - Hierarchical_regression.ipynb
  - Item_Response_Theory.ipynb
  - Neals_funnel.ipynb
  - Robust_Linear_Regression.ipynb
  - Sparse_Logistic_Regression.ipynb
  - Zero_inflated_count_data.ipynb

### Test plan

Before when tutorials had to use React style attributes, no styling
was applied see below.

using-react-style-in-jupyter

This was to ensure proper rendering in Docusaurus when the tutorial was
converted to MDX. Now we can use inline HTML style attributes in Jupyter
like below,

using-html-style-in-jupyter

and after the proposed changes the styles will be converted correctly to something
that React can consume.

correct-formatting-in-docusaurus

### Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](https://github.com/facebookresearch/beanmachine/blob/main/CONTRIBUTING.md)** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: website/scripts/convert_ipynb_to_mdx.py [hunk 1]
```diff
@@ -347,7 +356,7 @@ def transform_code_cell(  # noqa: C901 (flake8 too complex)
                             # markdown.
                             md = ""
                             if isinstance(md_df.index, pd.RangeIndex):
-                                md = md_df.to_markdown(showindex=False)
+                                md = md_df.to_markdown(index=False)
```
Comment (horizon-blue) [inline website/scripts/convert_ipynb_to_mdx.py:359]: Our internal `pandas` and `tabulate` packages are a bit old so they don't support the `index` argument :(

**Review by horizon-blue — COMMENTED** [review]
Thanks for working on this, Andy. Could you run `make format-notebook` to clear out the metadata from the `ipynb` files?

File: website/scripts/convert_ipynb_to_mdx.py (hunk 1 above)
Comment (ndmlny-qs) [reply website/scripts/convert_ipynb_to_mdx.py:359]: got it, will revert back

**Comment by ndmlny-qs** [comment]
I'm using `mdformat` again in this PR. I added it to `setup.py` to ensure the documenation build completes, but that does mean it overlaps with #1467

**Comment by horizon-blue** [comment]
:) Gotcha. Let me try importing this PR and rebasing on top of #1467 internally so we don't need to resolve merge conflict here

**Comment by ndmlny-qs** [comment]
awesome, I'm hopeful that this and a few other PRs will make tutorial writer's (and user's) experiences better

**Comment by ndmlny-qs** [comment]
rebased and now letting the tests run

**Pull request closed without merging**